File: RecyclerView backup/app/src/main/java/com/example/recyclerview/MainActivity.java
package com.example.recyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private RecyclerView.Adapter adapter;
    private RecyclerView.LayoutManager layoutManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ArrayList<RecyclerViewItem> recyclerViewItems = new ArrayList<>();
        recyclerViewItems.add(new RecyclerViewItem(R.drawable.baseline_sentiment_satisfied_alt_24,
                "Happy","Life is Qaiff!"));
        recyclerViewItems.add(new RecyclerViewItem(R.drawable.baseline_sentiment_neutral_24,
                "Normal","Life is Norm"));
        recyclerViewItems.add(new RecyclerViewItem(R.drawable.baseline_sentiment_very_dissatisfied_24,
                "Sad","Life is Suck!"));
        recyclerView = findViewById(R.id.testRecycleView);
        recyclerView.setHasFixedSize(true);
        adapter = new RecyclerViewAdapter(recyclerViewItems);
        layoutManager = new LinearLayoutManager(this);

        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(layoutManager);
    }
}

File: Magic8Ball/app/src/main/java/com/example/magic8ball/MainActivity.kt
package com.example.magic8ball

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import androidx.constraintlayout.widget.ConstraintLayout
import com.example.magic8ball.databinding.ActivityMainBinding
import kotlin.random.Random
import kotlin.random.nextInt

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(R.layout.activity_main)

        val view : ConstraintLayout = binding.root
        setContentView(view)

        val askMeButton: Button = findViewById(R.id.button)
//        val askMeButton : binding.button
        askMeButton.setOnClickListener { shakeBall() }
    }

    private fun shakeBall() {

        val ballImageView: ImageView = findViewById(R.id.imageView)

        val randomChoice = Random.nextInt(20) + 1
        val imageResource = when (randomChoice){

            1 -> R.drawable.magic_8_ball_1
            2 -> R.drawable.magic_8_ball_2
            3 -> R.drawable.magic_8_ball_3
            4 -> R.drawable.magic_8_ball_4
            5 -> R.drawable.magic_8_ball_5
            6 -> R.drawable.magic_8_ball_6
            7 -> R.drawable.magic_8_ball_7
            8 -> R.drawable.magic_8_ball_8
            9 -> R.drawable.magic_8_ball_9
            10-> R.drawable.magic_8_ball_10
            11-> R.drawable.magic_8_ball_11
            12-> R.drawable.magic_8_ball_12
            13-> R.drawable.magic_8_ball_13
            14-> R.drawable.magic_8_ball_14
            15-> R.drawable.magic_8_ball_15
            16-> R.drawable.magic_8_ball_16
            17-> R.drawable.magic_8_ball_17
            18-> R.drawable.magic_8_ball_18
            19-> R.drawable.magic_8_ball_19
            else -> R.drawable.magic_8_ball_20

        }

        ballImageView.setImageResource(imageResource)
    }
}